packages/ruleset/src/functions/schema-naming-convention.js

```diff
@@ -0,0 +1,481 @@
+/**
+ * Copyright 2023 IBM Corporation.
+ * SPDX-License-Identifier: Apache2.0
+ */
+
+const {
+  isObject,
+  schemaHasProperty,
+} = require('@ibm-cloud/openapi-ruleset-utilities');
+
+const {
+  LoggerFactory,
+  getRequestBodySchemaForOperation,
+  getResourceSpecificSiblingPath,
+  getResponseCodes,
+  getSuccessResponseSchemaForOperation,
+} = require('../utils');
+
+let ruleId;
+let logger;
+
+/**
+ * The implementation for this rule makes assumptions that are dependent on the
+ * presence of the following other rules:
+ * 
+ * - ibm-request-and-response-content: all request bodies and relevant success
+ *   responses define content objects
+ * 
+ * - ibm-content-contains-schema: the content objects define schemas
+ * 
+ * - ibm-avoid-inline-schemas: all schemas are named (defined with references)
+ * 
+ * - ibm-collection-array-property: the presence and correct name of the
+ *   property in a collection schema that holds the resource list
+ * 
+ * - (yet to be developed): schema names use upper camel case
```

Comment:
I'm working on this rule now - I'm going to push it to this same `api-symmetry` branch before I merge either of them, since they are inter-dependent